### PR TITLE
React task 1

### PR DESCRIPTION
**Pull request by mikitapapok**
# React task 1

## Info
  Create note list. Each element of the list shows: a title, first 20 symbols of a description and a date creation. Clicking  one of them it must be change its style. Active element information must display otherwise 'Select note to display' will be displayed.

### Requests.
1.Create a project with create-react-app.
2. Connect Prettier and ESLint.
3. use useState for keeping active note.
4. use prop-type library.
5. use material-ui, ant design for design.
6. Create project structure.

### Source
[React task 1](https://mikitapapok.github.io/react-task/)</br>

### CODE REVIEW

File: src/pages/myNotes/notes.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import React, { useState } from "react";
+import Box from "@mui/material/Box";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import NotesComponent from "./notesComponent";
+import ComponentInfo from "./componentInfo";
+const Notes = () => {
+  const [componentInfo, setComponentInfo] = useState(null);
+  const todoList = [
```
Comment (iTechArtSheps) [inline src/pages/myNotes/notes.jsx]: move to constants or mock data folder

File: src/pages/myNotes/componentInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          </p>
+          <p>
+            <span className="notes-list__component-text__component">Date:</span>{" "}
+            {props.item.date}
+          </p>
+        </article>
+      )}
+    </div>
+  );
+};
```
Comment (iTechArtSheps) [inline src/pages/myNotes/componentInfo.jsx:24]: add spaces

File: src/pages/myNotes/componentInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { number, string } from "prop-types";
+import PropTypes from "prop-types";
```
Comment (iTechArtSheps) [inline src/pages/myNotes/componentInfo.jsx]: add space

File: package.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
+{
+  "homepage": "https://mikitapapok.github.io/react-task/",
+  "name": "task-react",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@emotion/react": "^11.7.1",
```
Comment (iTechArtSheps) [inline package.json:7]: investigate

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+// module.exports = {
```
Comment (iTechArtSheps) [inline .eslintrc.js]: turn it on

File: src/pages/myNotes/notesComponent.jsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { ListItemText } from "@mui/material";
+import NotesComponentText from "./notesComponentText";
+import PropTypes, { objectOf } from "prop-types";
+const NotesComponent = (props) => {
```
Comment (iTechArtSheps) [inline src/pages/myNotes/notesComponent.jsx]: ({item, todo}) etc
____________
const {item, todo} = props

File: src/pages/myNotes/notesComponentText.jsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import PropTypes, { objectOf } from "prop-types";
+const NotesComponentText = (props) => {
+  const { id, title, date, description } = props.todo;
```
Comment (iTechArtSheps) [inline src/pages/myNotes/notesComponentText.jsx]: props?.todo

File: src/pages/myNotes/notes.jsx (hunk 1 above)
Comment (mikitapapok) [reply src/pages/myNotes/notes.jsx]: Moved

File: src/pages/myNotes/componentInfo.jsx (hunk 2 above)
Comment (mikitapapok) [reply src/pages/myNotes/componentInfo.jsx:24]: Added

File: src/pages/myNotes/componentInfo.jsx (hunk 3 above)
Comment (mikitapapok) [reply src/pages/myNotes/componentInfo.jsx]: Added

File: .eslintrc.js (hunk 5 above)
Comment (mikitapapok) [reply .eslintrc.js]: turned

File: src/constans/noteList.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
+export const todoList = [
+    {
```
Comment (iTechArtSheps) [inline src/constans/noteList.js]: a typo in the folder name -> constanTs

File: src/index.jsx [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import "./style/normalize.scss";
+import App from "./App.jsx";
```
Comment (iTechArtSheps) [inline src/index.jsx]: add space

File: src/pages/myNotes/componentInfo.jsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+
```
Comment (iTechArtSheps) [inline src/pages/myNotes/componentInfo.jsx:3]: remove space here, try to group all external dependencies together

File: src/pages/myNotes/noteListItem.jsx [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  showId,
+}) => {
+  return (
+    <div
+      className="notes-list__component-text"
+      onClick={getItemInfo ? getItemInfo : null}
+      id={id ? id : null}
+    >
+      <h2>{title}</h2>
+      {showId ? (
```
Comment (iTechArtSheps) [inline src/pages/myNotes/noteListItem.jsx]: change to `showId && {}`, ternary operator is redundant

File: src/pages/myNotes/noteListItem.jsx [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+NoteListItem.propTypes = {
+  id: PropTypes.number,
+  title: PropTypes.string,
+  description: PropTypes.string,
+  date: PropTypes.string,
+  getItemInfo: PropTypes.func,
+  showId: PropTypes.bool,
+};
```
Comment (iTechArtSheps) [inline src/pages/myNotes/noteListItem.jsx:36]: add space

File: src/pages/myNotes/notes.jsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
+import React, { useState } from "react";
+
+import { Container, NotesList, StyledListComponent } from "./styled";
```
Comment (iTechArtSheps) [inline src/pages/myNotes/notes.jsx]: move styled import at the end of this import list.

File: src/pages/myNotes/notes.jsx [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+
+const Notes = () => {
+  const [componentInfo, setComponentInfo] = useState(null);
+
+  const getItemInfo = (e) => {
+    const getCurrentItem = todoList.find((todo) => {
+      return todo.id == e.currentTarget.id;
+    });
+    setComponentInfo(getCurrentItem);
+    console.log(componentInfo);
```
Comment (iTechArtSheps) [inline src/pages/myNotes/notes.jsx]: remove console.log

File: src/pages/myNotes/notes.jsx [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+                    date={todo.date}
+                    getItemInfo={getItemInfo}
+                  />
+                }
+              />
+            </StyledListComponent>
+          ))}
+        </List>
+      </Container>
+      <ComponentInfo
```
Comment (iTechArtSheps) [inline src/pages/myNotes/notes.jsx]: just don't render the component, if you don't have data to do it.
{
componentInfo && <ComponentInfo {... all your props here with no conditions}/>
}

File: src/pages/myNotes/componentInfo.jsx [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+import { StyledComponentInfo } from "./styled";
+
+const ComponentInfo = ({ id, description, title, date }) => {
+  return (
+    <StyledComponentInfo className="active">
+      {!id ? (
+        <h1>Select note to display</h1>
+      ) : (
+        <NoteListItem
+          id={id ? id : null}
```
Comment (iTechArtSheps) [inline src/pages/myNotes/componentInfo.jsx]: remove the condition, 'id' should always exist

File: src/pages/myNotes/noteListItem.jsx [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+  title,
+  description,
+  date,
+  showId,
+}) => {
+  return (
+    <div
+      className="notes-list__component-text"
+      onClick={getItemInfo ? getItemInfo : null}
+      id={id ? id : null}
```
Comment (iTechArtSheps) [inline src/pages/myNotes/noteListItem.jsx]: same here

File: src/pages/myNotes/noteListItem.jsx [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+  id,
+  title,
+  description,
+  date,
+  showId,
+}) => {
+  return (
+    <div
+      className="notes-list__component-text"
+      onClick={getItemInfo ? getItemInfo : null}
```
Comment (iTechArtSheps) [inline src/pages/myNotes/noteListItem.jsx]: same here

File: src/pages/myNotes/notes.jsx [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+  };
+
+  return (
+    <NotesList>
+      <Container>
+        <List>
+          {todoList.map((todo) => (
+            <StyledListComponent
+              className={
+                componentInfo && componentInfo.id === todo.id ? "active" : null
```
Comment (iTechArtSheps) [inline src/pages/myNotes/notes.jsx]: do you really need the class name here? Why?

File: src/constans/noteList.js (hunk 8 above)
Comment (mikitapapok) [reply src/constans/noteList.js]: Renamed

File: src/index.jsx (hunk 9 above)
Comment (mikitapapok) [reply src/index.jsx]: Done

File: src/pages/myNotes/notes.jsx (hunk 15 above)
Comment (mikitapapok) [reply src/pages/myNotes/notes.jsx]: If I do this< Message 'Select note to display' won't be displayd

File: src/pages/myNotes/componentInfo.jsx (hunk 16 above)
Comment (mikitapapok) [reply src/pages/myNotes/componentInfo.jsx]: Done

File: src/pages/myNotes/noteListItem.jsx (hunk 17 above)
Comment (mikitapapok) [reply src/pages/myNotes/noteListItem.jsx]: Done

File: src/pages/myNotes/noteListItem.jsx (hunk 18 above)
Comment (mikitapapok) [reply src/pages/myNotes/noteListItem.jsx]: Fixed

File: src/pages/myNotes/notes.jsx (hunk 19 above)
Comment (mikitapapok) [reply src/pages/myNotes/notes.jsx]: To change style of clicked element

File: src/pages/myNotes/notes.jsx (hunk 14 above)
Comment (mikitapapok) [reply src/pages/myNotes/notes.jsx]: Removed

File: src/pages/myNotes/notes.jsx (hunk 13 above)
Comment (mikitapapok) [reply src/pages/myNotes/notes.jsx]: Moved

File: src/pages/myNotes/noteListItem.jsx (hunk 12 above)
Comment (mikitapapok) [reply src/pages/myNotes/noteListItem.jsx:36]: Added

File: src/pages/myNotes/noteListItem.jsx (hunk 11 above)
Comment (mikitapapok) [reply src/pages/myNotes/noteListItem.jsx]: Changed

File: src/pages/myNotes/componentInfo.jsx (hunk 10 above)
Comment (mikitapapok) [reply src/pages/myNotes/componentInfo.jsx:3]: Removed

File: src/pages/myNotes/notes.jsx (hunk 19 above)
Comment (iTechArtSheps) [reply src/pages/myNotes/notes.jsx]: I see, it makes sense and I can accept it, but let's try to do the same through props. I mean try to pass "isActive" prop to the component and in ./styled add styles based on the prop

Comment (mikitapapok) [reply src/pages/myNotes/notes.jsx]: Done

**Review by iTechArtSheps — APPROVED** [review]

**Review by iTechArtSheps — APPROVED** [review]

**Pull request merged**